src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1438,13 +1459,58 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         return terminalPosition;
     }
 
-    // clang-format off
+    // Method Description:
+    // - Checks if a double click occurred
+    // Arguments:
+    // - clickPos: the (x,y) position of a given cursor (i.e.: mouse cursor).
+    //    NOTE: origin (0,0) is top-left.
+    // - clickTime: the timestamp that the click occurred
+    // Return Value:
+    // - true if the new click was at the same location within the time delta
+    const bool TermControl::_IsDoubleClick(winrt::Windows::Foundation::Point clickPos, TimeStamp clickTime) const
+    {
+        if (clickPos == _lastMouseClickPos)
+        {
+            const TimeStamp threshold = 500000;
+            TimeStamp delta;
+            UInt64Sub(clickTime, _lastMouseClick, &delta);
+            if (delta < threshold)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // Method Description:
+    // - Checks if a triple click occurred
+    // Arguments:
+    // - clickPos: the (x,y) position of a given cursor (i.e.: mouse cursor).
+    //    NOTE: origin (0,0) is top-left.
+    // - clickTime: the timestamp that the click occurred
+    // Return Value:
+    // - true if the new click was at the same location within the time delta after a double click having occurred
+    const bool TermControl::_IsTripleClick(winrt::Windows::Foundation::Point clickPos, TimeStamp clickTime) const
+    {
+        if (_doubleClickOccurred && clickPos == _lastMouseClickPos)
+        {
+            const TimeStamp threshold = 500000;

```

Comment:
Could `500000` be a named constant? I have no idea how much time that represents

Reply:
Moved to `TermControl.h` w/ a comment. Good?

---
In reply to: [293364564](https://github.com/microsoft/terminal/pull/1197#discussion_r293364564) [](ancestors = 293364564)